harmonica/filters/_filters.py

```diff
@@ -62,3 +62,598 @@ def derivative_upward_kernel(fft_grid, order=1):
+        Array with the Fourier transform of the original grid.
+        Its dimensions should be in the following order:
+        *freq_northing*, *freq_easting*.
+        Use :func:`xrft.xrft.fft` and :func:`xrft.xrft.ifft` functions to
+        compute the Fourier Transform and its inverse, respectively.
+    inclination : float in degrees
+        The inclination inducing Geomagnetic field. Default is 90 degree for
+        RTP field.
+    declination : float in degrees
+        The declination inducing Geomagnetic field. Default is 0 degree for
+        RTP field.
+    magnetization_inclination : float in degrees
+        The inclination of the total magnetization of the anomaly source.
+        Default is i, neglecting remanent magnetization and
+        self demagnetization.
+    magnetization_declination : float in degrees
+        The declination of the total magnetization of the anomaly source.
+        Default is d, neglecting remanent magnetization and
+        self demagnetization.
+    f : float or 2d-array
+        Ambient field in the study area. It can use the mean ambinent field
+        value in the study area or the real ambient field value in all
+        locations. Default is 50,000 nT.
+
+    Returns
+    -------
+    da_filter : :class:`xarray.DataArray`
+        Array with the kernel for the pseudo gravity filter in frequency
+        domain.
+
+    References
+    ----------
+    [Salem2014]_
+
+    See also
+    --------
+    harmonica.pseudo_gravity
+    """
+    # Check if magnetization angles are valid
+    _check_magnetization_angles(magnetization_inclination, magnetization_declination)
+    # Define magnetization angles if they are None
+    if magnetization_declination is None and magnetization_inclination is None:
+        magnetization_inclination = inclination
+        magnetization_declination = declination
+    # Catch the dims of the Fourier transformed grid
+    dims = fft_grid.dims
+    # Grab the coordinates of the Fourier transformed grid
+    freq_easting = fft_grid.coords[dims[1]]
+    freq_northing = fft_grid.coords[dims[0]]
+    # Convert frequencies to wavenumbers
+    k_easting = 2 * np.pi * freq_easting
+    k_northing = 2 * np.pi * freq_northing
+    # Check if input is RTP field
+    if inclination == 90:
+        # Calculate vertical integral
+        da_filter = np.sqrt(k_easting**2 + k_northing**2) ** -1
+    else:
+        # Calculate RTP kernel, then calculate vertical integral
+        da_filter = (
+            _get_rtp_filter(
+                k_easting,
+                k_northing,
+                inclination,
+                declination,
+                magnetization_inclination,
+                magnetization_inclination,
+            )
+            * np.sqrt(k_easting**2 + k_northing**2) ** -1
+        )
+    # Deal with inf and nan value
+    da_filter.data = np.nan_to_num(da_filter.data, posinf=0, nan=0)
+    return da_filter / 149.8 / f
```

Comment:
The `149.8` sounds a little bit magical. Where does it come from? Is it the factor between the density and the magnitude of the magnetization? If that's the case we should include it as a parameter that could be changed. If it involves a universal constant, would be better to add it to the `harmonica.constants` module instead of hardcoding it here in the function.